### DISCUSSION TITLE
Postal doesn't go through the queue

### DISCUSSION THREAD

**Discussion by olegbliaher** [postalserver/postal#2531] [Help with using Postal]
Hi there,

We had one user that sent quite a big amount of email, and about 20,000 emails are still remaining in the queue. The workers are not retrieving any of the remaining queued emails.

Is there a way to force them to do so, for example via the console?

The emails are not held or suppressed.

I've tried to do the following on a few messages:

```
sender = cached_sender(Postal::SMTPSender, queued_message.message.recipient_domain, queued_message.ip_address)
sender.send_message(queued_message.message)
```

But that doesn't seem to be enough.

**Comment by willpower232** [comment]
You might need to add more workers with the scale argument, have a look at https://github.com/postalserver/postal/discussions/1461#discussioncomment-1119403

**Comment by olegbliaher** [comment]
Hi there, sorry, I should've mentioned that directly: I have 8 workers, and have been having that since a while. The server has 32 GB of RAM and 16 CPUs, so resources are not an issue. The issue is rather that the server isn't sending any of the emails that are lying in the queue.

**Reply by willpower232** [reply]
what is your `postal status`?

**Reply by olegbliaher** [reply]
`postal status` has been removed: 

https://github.com/postalserver/postal/commit/293ad9ee8c81dfe555ceb5ce75d8862bf88d2c8b#diff-db71f531b5e40d307384a9da25959bdb7f73f959a9c75722cd05d38fcdf871eeL46

**Reply by willpower232** [reply]
You should be using docker-compose which provides a status command https://github.com/postalserver/install/blob/ce3ddfa66e18a7509ebec9c7aebf84fcd70675df/bin/postal#L93-L95

You shouldn't need to execute commands inside any of the containers.

**Reply by olegbliaher** [reply]
Ok, it gets slightly more complicated here: I am running an un-dockerized Postal 2.1.4 on a Linux-server. 

RabbitMQ and the workers are on. Newer queues are being handled, but not the queues created on the 7/7 (the day I started this thread).

Is it the workers or the cron telling Postal to pick up messages from the queue?

**Reply by olegbliaher** [reply]
Could it be that a restart of RabbitMQ messed the queue-handling up?

**Reply by willpower232** [reply]
The queue workers lock the message when it is being handled in some capacity so it is possible some messages are still locked in your database, you will have to have a rummage around. Running postal without docker is quite tricky and definitely not very easy to support.

**Reply by olegbliaher** [reply]
Hi @willpower232, that was very helpful. This is a well-used server that has been up for quite a while, so I believe there could definitely be plenty of cleaning up to do in the database, and the `locked_at` column was a very good clue to how the workers behave.

I will also set up the RabbitMQ management tool, would be very interesting to get some insight on how the queues _actually_ get handled.
